Project2015To2017.Migrate2017.Tool/CommandLogic.cs

```diff
@@ -1,89 +1,198 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
+using DotNet.Globbing;
 using Project2015To2017.Analysis;
 using Project2015To2017.Definition;
+using Project2015To2017.Reading;
 using Project2015To2017.Transforms;
 using Project2015To2017.Writing;
+using Serilog;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
 
 namespace Project2015To2017.Migrate2017.Tool
 {
 	public class CommandLogic
 	{
 		private readonly Microsoft.Extensions.Logging.ILogger genericLogger;
+		private ImmutableArray<(string path, string extension)> files;
+		private readonly ImmutableArray<string> extensions;
 
 		public CommandLogic()
 		{
 			this.genericLogger = new Serilog.Extensions.Logging.SerilogLoggerProvider().CreateLogger(nameof(Serilog));
+			extensions = ProjectConverter.ProjectFileMappings.Keys.Concat(new[] {".sln"}).ToImmutableArray();
 		}
 
-		public IReadOnlyCollection<Project> ParseProjects(
+		public void DoProcessableFileSearch(bool force = false)
+		{
+			if (files != null && files.Length > 0 && !force)
+			{
+				Log.Verbose("Glob file list reevaluation skipped");
+				return;
+			}
+
+			Log.Verbose("Glob file list reevaluation started");
+			files = Directory.EnumerateFiles(Directory.GetCurrentDirectory(), "*.*", SearchOption.AllDirectories)
+				.Select(x => (path: x, extension: Path.GetExtension(x)?.ToLowerInvariant()))
+				.Where(x => !string.IsNullOrEmpty(x.extension))
+				.Where(x => extensions.Contains(x.extension))
+				.ToImmutableArray();
+			Log.Verbose("Glob file list reevaluation finished: {Count} items", files.Length);
+		}
+
+		public (IReadOnlyCollection<Project> projects, IReadOnlyCollection<Solution> solutions) ParseProjects(
 			IEnumerable<string> items,
 			ITransformationSet transformationSet,
 			ConversionOptions conversionOptions)
 		{
+			var converter = new ProjectConverter(genericLogger, transformationSet, conversionOptions);
 			var convertedProjects = new List<Project>();
+			var convertedSolutions = new List<Solution>();
 
-			foreach (var file in items)
+			foreach (var pattern in items)
 			{
-				var projects = new ProjectConverter(genericLogger, transformationSet, conversionOptions)
-					.Convert(file)
-					.Where(x => x != null)
-					.ToList();
-				convertedProjects.AddRange(projects);
+				if (File.Exists(pattern))
+				{
+					var file = new FileInfo(pattern);
```

Comment:
We should handle/rewrite the slashes here so we don't introduce new cases of #176